brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/Alien4CloudFacade.java

```diff
@@ -628,46 +593,37 @@ Operation getOperation() {
         }
     }
 
-    class RelationshipGenerator extends ScriptGenerator {
+    class RelationshipScriptGenerator extends ScriptGenerator {
 
-        private String requirementId;
+        private ToscaApplication.Relationship relationship;
         private RelationshipTemplate relationshipTemplate;
 
-        public RelationshipGenerator(String opKey, String nodeId, Alien4CloudApplication toscaApplication, String requirementId, String computeName, String expandedFolder) {
-            super(opKey, nodeId, toscaApplication, computeName, expandedFolder);
-            this.requirementId = requirementId;
+        public RelationshipScriptGenerator(String opKey, Alien4CloudApplication toscaApplication, ToscaApplication.Relationship relationship, String computeName, String expandedFolder) {
+            super(opKey, relationship.getSourceNodeId(), toscaApplication, computeName, expandedFolder);
+            this.relationship = relationship;
         }
 
         @Override
         Object getScript(ImplementationArtifact artifact, Operation op) {
-            Optional<RelationshipTemplate> optionalRelationshipTemplate = findRelationshipRequirement(nodeId, toscaApplication, requirementId);
+            Optional<RelationshipTemplate> optionalRelationshipTemplate = Optional.fromNullable(toscaApplication.getNodeTemplate(nodeId).getRelationships().get(relationship.getRelationshipId()));
             if (!optionalRelationshipTemplate.isPresent()) {
                 LOG.warn("Unsupported operation implementation for " + op.getDescription() + ": no relationship template");
-                return Optional.absent();
+                return null;
             }
             this.relationshipTemplate = optionalRelationshipTemplate.get();
 
             String script = getScript(artifact);
-            return buildExportStatements(op, script).or(script);
+            return Optional.of(buildExportStatements(op, script).or(script));
```

Comment:
Optional of Optional should just be `buildExportStatements(op, script).or(script)`